### PR TITLE
refactor(router): pass NamespaceId through handler stack

### PR DESCRIPTION
**Pull request by domodwyer**
The first in a series of PRs to push namespace + table IDs over Kafka (#4880).

This PR changes the router to propagate the `NamespaceId` through the request handlers, so it is available when serialising the DML operation into the Kafka wire format.

---

* style: format imports (0c5eb3f70)

      Re-order and re-format the imports so that they follow a consistent
      pattern.
      
      This helps eliminate conflicts due to imports.

* refactor: resolve namespace before DML dispatch (d166de931)

      This commit introduces a new (composable) trait; a NamespaceResolver is
      an abstraction responsible for taking a string namespace from a user
      request, and mapping to it's catalog ID.
      
      This allows the NamespaceId to be injected through the DmlHandler chain
      in addition to the namespace name.
      
      As part of this change, the NamespaceAutocreation layer was changed from
      an implementator of the DmlHandler trait, to a NamespaceResolver as it
      is a more appropriate abstraction for the functionality it provides.

### CODE REVIEW

File: router/src/lib.rs [hunk 1]
```diff
@@ -1,14 +1,96 @@
...
+//!                           ┌──────────────┐
+//!                           │ Write Buffer │
+//!                           └──────────────┘
+//!                                   │
+//!                                   │
+//!                          ┌────────▼─────┐
+//!                          │    Kafka     ├┐
+//!                          └┬─────────────┘├┐
+//!                           └┬─────────────┘│
+//!                            └──────────────┘
```
Comment (domodwyer) [inline router/src/lib.rs:70]: This documents the structure of the router, and the various handlers a request progresses through.

File: router/src/lib.rs [hunk 2]
```diff
@@ -1,14 +1,96 @@
...
+//!                      ║  │   Autocreation   │ ─║─ ─ ─ ─ ─ ─
+//!                      ║  └──────────────────┘  ║           │
+//!                      ║            │           ║
+//!                      ║            ▼           ║           │
+//!                      ║  ┌──────────────────┐  ║
+//!                      ║  │ NamespaceSchema  │  ║           │
+//!                      ║  │     Resolver     │ ─║─ ─ ─ ─ ─ ─
+//!                      ║  └──────────────────┘  ║           │
+//!                      ║            │           ║
+//!                      ╚════════════│═══════════╝           │
```
Comment (domodwyer) [inline router/src/lib.rs:37]: This is the subject of this PR - the namespace autocreator has moved from a `DmlHandler` to a `NamespaceResolver` and a new `NamespaceSchemaResolver` uses the namespace cache to discover the `NamespaceId` for the request without querying the catalog.

File: router/src/namespace_resolver.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                // overwrites, but an entry will always exist once inserted, and
+                // the schemas will eventually converge.
+                self.cache
+                    .put_schema(namespace.clone(), Arc::clone(&schema));
+
+                trace!(%namespace, "schema cache populated");
+                Ok(schema.id)
+            }
+        }
+    }
```
Comment (domodwyer) [inline router/src/namespace_resolver.rs:97]: This is the entirety of the new logic this PR adds - the rest of the diff is moving things around.

**Review by mkmik — APPROVED** [review]
love the diagrams

**Pull request merged**